client/mysql.cc

```diff
@@ -5315,6 +5365,512 @@ static int com_prompt(String *buffer __attribute__((unused)),
+                                      name_len);
+      if (alias)
+      {
+        tee_fprintf(stdout, "alias  %s = '%s'\n", alias->name, alias->value);
+      }
+      else
+      {
+        tee_fprintf(stdout, "alias: '%s': not found\n", name);
+      }
+    }
+    *error= false;
+  }
+  return pos;
+}
+
+static char *handle_next_unalias(char *line, bool *error)
+{
+  char *name, *pos;
+  uchar *record;
+  bool is_valid;
+
+  /* Parse the alias name. */
+  pos= parse_alias_name(line, &name, &is_valid);
+
+  if (!is_valid)
+  {
+    tee_fprintf(stdout, "unalias: '%s': invalid alias name\n", name);
+    *error= true;
+  }
+  else
+  {
+    if ((record= my_hash_search(&aliases, (const uchar *) name, strlen(name))))
+    {
+      my_hash_delete(&aliases, record);
+    }
+    else
+    {
+      tee_fprintf(stdout, "unalias: '%s': not found\n", name);
+    }
+    *error= false;
+  }
+  return pos;
+}
+static int init_alias()
+{
+  FILE *file= 0;
+  char *mariadbrc_file;
+  char *line= 0;
+  char *ptr;
+  size_t len= 0;
+  ssize_t read;
+  MY_STAT stat_arg;
+  bool error;
+  int pos;
+
+  /* Initialize the hash structure to store aliases. */
+  if (my_hash_init2(&aliases, 64, charset_info,
+                    64, 0, 0, get_alias_key, 0,
+                    alias_free, MYF(0)))
+  {
+    put_info("Couldn't initialize hash to store aliases.", INFO_ERROR);
+    exit(1);
+  }
+
+  /*
+    Read alias commands from mariadbrc file.
+  */
+  if (getenv("MARIADBRC_FILE"))
+  {
+    mariadbrc_file= my_strdup(getenv("MARIADBRC_FILE"), MYF(MY_WME));
+  }
+  else if (getenv("HOME"))
+  {
+    mariadbrc_file= (char*) my_malloc((uint) strlen(getenv("HOME"))
+                                      + (uint) strlen("/.mariadbrc")
+                                      + 2, MYF(MY_WME));
+    if (mariadbrc_file)
+      sprintf(mariadbrc_file, "%s/.mariadbrc", getenv("HOME"));
+  }
+
+  if (mariadbrc_file == 0)
+  {
+    put_info("out-of-memory!", INFO_ERROR);
+    exit(1);
+  }
+
+  /* Check if mariadbrc file exists. */
+  if (!my_stat(mariadbrc_file, &stat_arg, MYF(MY_WME)))
+    goto cleanup;                               /* Do nothing */
+
+  if (!(file= my_fopen(mariadbrc_file, O_RDONLY, MYF(MY_WME))))
+  {
+    put_info("Failed to open .mariadbrc file: ", INFO_ERROR);
+    put_info(strerror(errno), INFO_ERROR, errno);
+    exit(1);
+  }
+
+  while ((read= getline(&line, &len, file)))
+  {
+    if (read == -1)                             /* EOF/Error */
+    {
+      if (errno == 0)
+        break;
+
+      put_info("Failed to read from .mariadbrc file: ", INFO_ERROR);
+      put_info(strerror(errno), INFO_ERROR, errno);
+      exit(1);
+    }
+
+    /* Remove trailing newline character. */
+    pos= read;
+    while (--pos && pos > 0 && my_iscntrl(charset_info, line[pos]))
+      line[pos]= 0;
+
+    ptr= line;
+
+    /* Remove leading whitespaces. */
+    while (*ptr && my_isspace(charset_info, *ptr))
+      ptr ++;
```

Comment:
```ptr++```, ```ptr+=``` here and everywhere else please.

Since ```'\0'``` is not a space, you don't need to explicitly check for it.